textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/common/KeywordPredicate.java

```diff
@@ -87,5 +119,10 @@ public Analyzer getAnalyzer(){
         return analyzer;
     }
 
+    public DataReaderPredicate convertToDataReaderPredicate() {
+        DataReaderPredicate dataReaderPredicate = new DataReaderPredicate(this.dataStore,this.queryObject);
```

Comment:
predicate is of type keyword predicate, which need to be converted to DataReaderPredicate